processor/src/range/processors.rs

```diff
@@ -0,0 +1,124 @@
+use super::{Felt, FieldElement};
+use crate::trace::LookupTableRow;
+
+// PROCESSOR RANGE CHECKS
+// ================================================================================================
+
+/// A struct containing all the range check lookups requested by user operations at a single clock
+/// cycle, grouped by the processor performing the lookup.
+#[derive(Debug, Clone)]
+pub struct CycleRangeChecks {
+    memory: Option<ProcessorRangeChecks>,
+    stack: Option<ProcessorRangeChecks>,
+}
```

Comment:
I left a note in the `aux_trace` file instead, because I think we'll actually want to rework the `AuxTraceBuilder` to track hints and rows (like in the other AuxTraceBuilders) rather than using the `BTreeMap`.